presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSkipEmptyFiles.java

```diff
@@ -0,0 +1,392 @@
+        if (!created) {
+            throw new RuntimeException("Could not create resource directory: " + temporaryDirectory.getPath());
+        }
+        @Language("SQL") String createQuery = format("CREATE TABLE %s.\"%s\".\"%s\" (id %s, field %s) WITH (external_location = '%s'," +
+                        "format = 'Parquet',partitioned_by = ARRAY['field']," +
+                        "  bucketed_by = ARRAY['id']," +
+                        "  bucket_count = 3)",
+                CATALOG, SCHEMA, tableName, IntegerType.INTEGER, VarcharType.VARCHAR, getResourceUrl(tableName));
+        queryRunner.execute(createQuery);
+        String partitionDirectoryPath = temporaryDirectory.getPath() + "/field=field1";
+        String name = "field";
+        @Language("SQL") String insertQuery;
+        for (int i = 1; i <= 5; i++) {
+            insertQuery = format("INSERT INTO %s.\"%s\".\"%s\" VALUES (%s,'%s')",
+                    CATALOG, SCHEMA, tableName, i, name + i);
+            queryRunner.execute(insertQuery);
+            if (i == 1) {
+                File secondParquetFile = new File(partitionDirectoryPath, randomUUID().toString());
+                if (!secondParquetFile.createNewFile()) {
+                    throw new RuntimeException("Could not create empty file");
+                }
+            }
+        }
+        if (replace) {
+            File partitionDirectory = new File(partitionDirectoryPath);
+            deleteMetadata(partitionDirectory, true);
+        }
+        return temporaryDirectory;
+    }
+
+    /**
+     * Deletes the given directory and all of its contents recursively
+     * Does not follow symbolic links
+     *
+     * @param temporaryDirectory a {@link File} pointing to the directory to delete
+     * @param hasPattern a {@code true} if it is necessary to delete only one directory file, {@code false} otherwise
+     */
+    private static void deleteMetadata(File temporaryDirectory, boolean hasPattern)
+    {
+        @Language("RegExp") String nameStart = "000000_.*";
+        File[] metadataFiles = temporaryDirectory.listFiles();
+        if (metadataFiles != null) {
+            for (File file : metadataFiles) {
+                if ((!Files.isSymbolicLink(file.toPath()) && !hasPattern) || (hasPattern && file.getName().matches(nameStart))) {
+                    deleteMetadata(file, false);
+                }
+            }
+        }
+        if (!hasPattern && !temporaryDirectory.delete()) {
+            throw new RuntimeException("Could not to delete metadata");
+        }
+    }
+
+    @Test
+    public void testSkipEmptyFilesSuccessful()
+            throws Exception
+    {
+        String tableName = "skip_empty_files_success";
+        File resourcesLocation = generateMetadata(tableName);
+        executeCreationTestAndDropCycle(queryRunner, tableName, getResourceUrl(tableName), false, null);
+        deleteMetadata(resourcesLocation, false);
```

Comment:
deletion in tests should be handled in an after test method to ensure cleanup in the face of exceptions

Reply:
Done